src/windows-emulator/syscalls.cpp

```diff
@@ -3413,6 +3413,31 @@ namespace
         c.win_emu.yield_thread();
         return STATUS_SUCCESS;
     }
+    typedef UNICODE_STRING<EmulatorTraits<Emu64>> *PUNICODE_STRING;
+    typedef struct _CLSMENUNAME
+    {
+        LPSTR pszClientAnsiMenuName;
+        LPWSTR pwszClientUnicodeMenuName;
+        PUNICODE_STRING pusMenuName;
+    } CLSMENUNAME, *PCLSMENUNAME;
+    NTSTATUS handle_NtUserRegisterClassExWOW(
+        const syscall_context& c, const emulator_object<WNDCLASSEXW> lpwcx,
+        const emulator_object<UNICODE_STRING<EmulatorTraits<Emu64>>> pustrClassName,
+        const emulator_object<UNICODE_STRING<EmulatorTraits<Emu64>>> ClsNVersion,
+        const emulator_object<CLSMENUNAME> pClassMenuName)
+    {
+
+        UNICODE_STRING<EmulatorTraits<Emu64>> ClassName = pustrClassName.read();
+
+        const auto allocatedAddress = c.emu.allocate_memory(0x1000, memory_permission::read_write);
+
+        emulator_object<RTL_ATOM> atom{c.emu, allocatedAddress};
+
+        handle_NtAddAtomEx(c, ClassName.Buffer, ClassName.Length, atom, /*Flags=*/0);
+
+        return atom.read();
```

Comment:
this is not exactly what i meant. my intent was rather to extract common parts. like what both functions do is they loop through all existing atoms, to check if it already exists, or they add a new one. so you could extract that into a `uint16_t store_atom(const syscall_context& c, std::u16string atom)` function and just call that one from both syscall handlers